src/patchers/SwitchCasePatcher.js

```diff
@@ -0,0 +1,86 @@
+import NodePatcher from './NodePatcher.js';
+import type { Editor, Node, ParseContext, Token } from './types.js';
+
+export default class SwitchCasePatcher extends NodePatcher {
+  constructor(node: Node, context: ParseContext, editor: Editor, conditions: NodePatcher, consequent: NodePatcher) {
+    super(node, context, editor);
+    this.conditions = conditions;
+    this.consequent = consequent;
+    this.negated = false;
+  }
+
+  patchAsStatement() {
+    let { conditions, consequent } = this;
```

Comment:
I've been moving away from this pattern and just accessing properties on `this` as needed.